### PR TITLE
Cleanin' up some stuff

### PR DESCRIPTION
**Pull request by michaelbprice**
#### PR Classification
Code cleanup and test enhancement.

#### PR Summary
Reformatted structs, removed unnecessary includes, ensured memory alignment, and added new test cases.
- `Input.h`: Reformatted `Axis` struct within `JoystickInfo` and added 'f' to float literals.
- `SystemEditor.cpp`: Removed several `#include` directives.
- `StarSystem.h`: Added `#pragma pack` directives for memory alignment.

Removed a demo file.

### CODE REVIEW

File: src/Input.h [hunk 1]
```diff
@@ -76,12 +76,12 @@ namespace Input {
 	};
 
 	struct JoystickInfo {
-		struct Axis {
-			float value = 0.0;
-			float deadzone = 0.1;
-			float curve = 1.0;
-			bool zeroToOne = false;
-		};
+        struct Axis {
```
Comment (MryamGirmay) [inline src/Input.h:79]: Thank you for resolving the initialization warning.

File: src/galaxy/StarSystem.h [hunk 2]
```diff
@@ -25,6 +25,7 @@ class CustomSystem;
 // doubles - all masses in Kg, all lengths in meters
 // fixed - any mad scheme
 
+#pragma pack(push, 4)
```
Comment (MryamGirmay) [inline src/galaxy/StarSystem.h:28]: Great job on reducing memory padding here.

Comment (michaelbprice) [reply src/galaxy/StarSystem.h:28]: I agree. Excllent.

File: src/Input.h (hunk 1 above)
Comment (michaelbprice) [reply src/Input.h:79]: It was my pleasure.

**Pull request closed without merging**